server/routes.go

```diff
@@ -940,6 +942,11 @@ func (s *Server) CreateBlobHandler(c *gin.Context) {
 		c.Status(http.StatusOK)
 		return
 	}
+	if c.GetHeader("X-Redirect-Create") == "1" && s.isLocal(c) {
```

Comment:
We can simplify this: instead of the client having to copy the file, let's make the server read it directly. The reason this is important:

* Multi-file input (e.g. safetensor files) will be much faster if the server can read them directly
* What if the cli can't access the model directory due to permissions? (e.g. on Linux) the server will end up having the final say as to whether a file is safely read

Why don't we try this:

1. Client makes a `POST /api/blobs/<blob sha>` with the body as today. However it also provides a header `X-File-Location` with the absolute path to the file.
2. The server first tries to read this path, and responds immediately with a 204 if its created

Reply:
Hey Jeff, I've done the server-side copy from the server in this PR here: https://github.com/ollama/ollama/pull/5887/files